cpp/oneapi/dal/table/backend/convert.hpp

```diff
@@ -55,6 +67,18 @@ void convert_vector(const detail::data_parallel_policy& policy,
                     std::int64_t dst_stride,
                     std::int64_t element_count);
 
+void convert_matrix(const detail::data_parallel_policy& policy,
```

Comment:
It would be nice to add a description here that this function might optionally copy data to device in some cases. And to describe in which cases it does so.